test/foundry/unit/concrete/modulemanager/TestModuleManager_FallbackHandler.t.sol

```diff
@@ -290,20 +289,120 @@ contract TestModuleManager_FallbackHandler is TestModuleManagement_Base {
     }
 
     function test_onTokenReceived_Success() public {
-        vm.startPrank(address(ENTRYPOINT));
+        erc721.safeMint(address(BOB_ACCOUNT), 1);
+        assertEq(erc721.balanceOf(address(BOB_ACCOUNT)), 1);
+
+        erc721.safeMint(ALICE_ADDRESS, 2);
+        vm.prank(ALICE_ADDRESS);
+        erc721.safeTransfer(address(BOB_ACCOUNT), 2);
+        assertEq(erc721.ownerOf(2), address(BOB_ACCOUNT));
+
+        erc1155.safeMint(address(BOB_ACCOUNT), 1, 1);
+        assertEq(erc1155.balanceOf(address(BOB_ACCOUNT), 1), 1);
+
+        uint256[] memory tokenIds = new uint256[](2);
+        tokenIds[0] = 2;
+        tokenIds[1] = 3;
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = 10;
+        amounts[1] = 30;
+        erc1155.safeMintBatch(address(BOB_ACCOUNT), tokenIds, amounts);
+
+        assertEq(erc1155.balanceOf(address(BOB_ACCOUNT), 2), 10);
+        assertEq(erc1155.balanceOf(address(BOB_ACCOUNT), 3), 30);
+
         //ERC-721
         (bool success, bytes memory data) = address(BOB_ACCOUNT).call{value: 0}(hex'150b7a02');
         assertTrue(success);
-        assertTrue(keccak256(data) == keccak256(bytes(hex'150b7a02')));
```

Comment:
wondering how did this pass before if the data it was returning was just a free mem pointer

Reply:
it was not a free memory pointer
it was a correct data just it was returned as bytes memory instead of bytes4